src/circuit/server/advert-service.ts

```diff
@@ -0,0 +1,108 @@
+import { logger } from '@libp2p/logger'
+import { codes } from '../../errors.js'
+import { namespaceToCid } from '../utils.js'
+import {
+  DEFAULT_ADVERT_BOOT_DELAY,
+  RELAY_RENDEZVOUS_NS
+} from '../constants.js'
+import type { Startable } from '@libp2p/interfaces/startable'
+import type { ContentRouting } from '@libp2p/interface-content-routing'
+import { CustomEvent, EventEmitter } from '@libp2p/interfaces/events'
+import pRetry from 'p-retry'
+
+const log = logger('libp2p:circuit-relay:advert-service')
+
+export interface AdvertServiceInit {
+  /**
+   * How long to wait after startup to begin advertising the service
+   * - if some configured content routers take a while to warm up (for
+   * example, the DHT needs some peers to be able to publish) this
+   * value should be high enough that they will have warmed up
+   */
+  bootDelay?: number
```

Comment:
seems like a hack, no? Should the content routers know to hold off on publishing until they are warm enough? This component would have less context on that than the content router itself.